autometa/taxonomy/ncbi.py

```diff
@@ -364,28 +338,81 @@ def parse_merged(self):
             logger.debug('merged loaded')
         return merged
 
+    def get_lineage_names(self, taxid):
+        """
+        Returns the lineage names of the taxid
+
+        Parameters
+        ----------
+        taxid : int
+            identifer for a taxon in the Taxonomy Database by NCBI
+
+        Returns
+        -------
+        set
+            lineage of the taxid
+        """
+        lineage = set()
+        while taxid != 1:
+            if self.rank(taxid) not in NCBI.CANONICAL_RANKS:
+                taxid = self.parent(taxid)
+                continue
+            lineage.add(self.name(taxid))
+            taxid = self.parent(taxid)
```

Comment:
You could avoid the same line `taxid = self.parent(taxid)` and remove the `continue` here by flipping some logic

```suggestion
            if self.rank(taxid) in NCBI.CANONICAL_RANKS:
                lineage.add(self.name(taxid))
            taxid = self.parent(taxid)
```